llama-rs/src/loader2.rs

```diff
@@ -110,9 +110,9 @@ impl<F: FnMut(LoadProgress)> ggml_loader::LoadHandler<LoadError, BufReader<&File
             Ok(id) => id,
             Err(err) => return ControlFlow::Break(LoadError::InvalidIntegerConversion(err)),
         };
-        self.vocab
-            .push_token(id, token, score)
-            .expect("vocab should be valid");
+        if let Err(err) = self.vocab.push_token(id, token, score) {
```

Comment:
again, given the API usage, this is impossible

Reply:
Hm. Sure. It mattered more when there was an error for duplicate tokens, but in this case it would be a breaking of the loading invariants if we encounter an ID out of step. I'll change that back to a panic.